lib/go/test/flow_epoch_test.go

```diff
@@ -1278,4 +1274,93 @@ func TestEpochReset(t *testing.T) {
 		result = executeScriptAndCheck(t, b, templates.GenerateGetQCEnabledScript(env), nil)
 		assert.Equal(t, cadence.NewBool(false), result)
 	})
+
+	result := executeScriptAndCheck(t, b, templates.GenerateGetFlowTotalSupplyScript(env), nil)
+	assertEqual(t, CadenceUFix64("7000000000.0"), result)
+
+	t.Run("Can reset the epoch during the staking auction with automatic rewards enabled", func(t *testing.T) {
+
+		tx := createTxWithTemplateAndAuthorizer(b, templates.GenerateEpochSetAutomaticRewardsScript(env), idTableAddress)
+
+		_ = tx.AddArgument(cadence.NewBool(true))
+
+		signAndSubmit(
+			t, b, tx,
+			[]flow.Address{idTableAddress},
+			[]crypto.Signer{IDTableSigner},
+			false,
+		)
+
+		var startView uint64 = 100
+		var stakingEndView uint64 = 120
+		var endView uint64 = 160
+
+		tx = createTxWithTemplateAndAuthorizer(b, templates.GenerateResetEpochScript(env), idTableAddress)
+		tx.AddArgument(cadence.NewUInt64(startEpochCounter + 1))
+		tx.AddArgument(CadenceString("stillSoRandom"))
+		tx.AddArgument(cadence.NewUInt64(startView))
+		tx.AddArgument(cadence.NewUInt64(stakingEndView))
+		tx.AddArgument(cadence.NewUInt64(endView))
+
+		signAndSubmit(
+			t, b, tx,
+			[]flow.Address{idTableAddress},
+			[]crypto.Signer{IDTableSigner},
+			false,
+		)
+
+		// Verify the current epoch's metadata to make sure rewards were calculated
+		// properly after the reset and that they haven't been paid yet
+		verifyEpochMetadata(t, b, env,
+			EpochMetadata{
+				counter:        startEpochCounter + 2,
+				seed:           "stillSoRandom",
+				startView:      startView,
+				endView:        endView,
+				stakingEndView: stakingEndView,
+				// The calculation of the total rewards should have happened
+				// because automatic rewards are enabled
+				// (total supply + current payount amount - bonus tokens) * reward increase factor
+				// (7000000000 + 1250000 - 0) * 0.00093871 = 6,571,204.6775
+				totalRewards:          "6572143.3875",
+				rewardsBreakdownArray: 0,
+				rewardsPaid:           false,
+				collectorClusters:     nil,
+				clusterQCs:            nil,
+				dkgKeys:               nil})
+
+		tx = createTxWithTemplateAndAuthorizer(b, templates.GenerateEpochPayRewardsScript(env), idTableAddress)
+
+		signAndSubmit(
+			t, b, tx,
+			[]flow.Address{idTableAddress},
+			[]crypto.Signer{IDTableSigner},
+			false,
+		)
+
+		// Verifies that the rewards from the previous epoch does not include the new epoch's amount
```

Comment:
When we spork, the rewards for the epoch pre-spork will be paid out when the `resetEpoch` transaction is submitted. Currently this is the next day, on Thursday. It would be ideal if this multisig could be moved to Wednesday, so there isn't a 1-day delay in reward payouts, once this is enabled. cc @vishalchangrani

Reply:
Agreed, it would be nice to reset on the same day as the spork